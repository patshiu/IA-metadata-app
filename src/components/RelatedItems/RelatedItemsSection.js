import React from 'react';
import PropTypes from 'prop-types';

import './RelatedItemSection.scss';

import SectionTitle from '../SectionTitle/SectionTitle';
import RelatedItem from './RelatedItem';

const RelatedItemsSection = ({ relatedItems }) => {
  if (!relatedItems || relatedItems.length === 0) {
    return (
      <div className="row">
        <div className="section">
          <SectionTitle title="Related Items" count={null} />
          <p>No related items.</p>
        </div>
      </div>
    );
  } else {
    return (
      <div className="row">
        <div className="section">
          <SectionTitle title="Related Items" count={relatedItems.length} />
          <div className="related-cards-container">
            {relatedItems.map((item, index) => {
              return (
                <RelatedItem key={index} itemSource={item._source} itemIdentifier={item._id} />
              );
            })}
          </div>
        </div>
      </div>
    );
  }
};

RelatedItemsSection.propTypes = {
  relatedItems: PropTypes.array
};

export default RelatedItemsSection;
