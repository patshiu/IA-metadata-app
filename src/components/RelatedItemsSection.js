import React from 'react';
import PropTypes from 'prop-types';

import './RelatedItemSection.scss';

import SectionTitle from './SectionTitle';
import RelatedItem from './RelatedItem';

const RelatedItemsSection = ({ relatedItems }) => {
  if (!relatedItems) {
    return (
      <div className="row">
        <div className="section">
          <SectionTitle title="Related Items" count={null} />
          <p>No related items.</p>
        </div>
      </div>
    );
  } else {
    return (
      <div className="row">
        <div className="section">
          <SectionTitle title="Related Items" count={relatedItems.length} />
          <div className="related-cards-container">
            {relatedItems.map((item, index) => {
              return (
                <RelatedItem key={index} itemSource={item._source} itemIdentifier={item._id} />
              );
            })}
          </div>
        </div>
      </div>
    );
  }
};

RelatedItemsSection.propTypes = {
  relatedItems: PropTypes.array
};

export default RelatedItemsSection;
