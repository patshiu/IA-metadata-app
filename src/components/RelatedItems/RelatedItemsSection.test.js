import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import RelatedItemsSection from './RelatedItemsSection';

it('renders review section with no reviews correctly', () => {
  const relatedItems = [];
  const tree = renderer.create(<RelatedItemsSection relatedItems={relatedItems} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders review section with undefined reviews object correctly', () => {
  const relatedItems = undefined;
  const tree = renderer.create(<RelatedItemsSection relatedItems={relatedItems} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders related items section with several items correctly', () => {
  const relatedItems = [
    {
      _score: 20.881666,
      _type: '_doc',
      _id: 'ToTheFairB',
      _source: {
        description: [
          "Provides a trip to the New York World's Fair of 1965, showing the various activities at the fair."
        ],
        title: ['To The Fair (Pt 2)'],
        downloads: [14757],
        mediatype: ['movies'],
        collection: [
          'prelinger',
          'fav-rubyrichardson',
          'fav-okay_amad',
          'fav-edwardkildow',
          'fav-made42',
          'fav-chi_ly',
          'fav-annelovesspam',
          'fav-neovict',
          'fav-lsofmo',
          'fav-skg22',
          'fav-captainrotundo',
          'fav-agnes_czes',
          'fav-foolishblue',
          'fav-yolandia',
          'fav-palmettogirl',
          'fav-mirkaelina',
          'fav-telos_in_motion',
          'fav-barria',
          'fav-e_scott_pierce',
          'fav-mikegiannoni',
          'fav-peterttbb',
          'fav-lakislalakis',
          'fav-drewbus128',
          'fav-victorhelice',
          'fav-chrbir5',
          'fav-sgtremp',
          'fav-theorasmussen',
          'fav-shilgiah',
          'fav-rd1234',
          'fav-green_arrow_green_lantern',
          'fav-astral_void',
          'fav-n_black',
          'fav-lois487',
          'fav-man_roach',
          'fav-samantha_bahlmann',
          'fav-metrotoronto',
          'fav-applesquash',
          'fav-philly_walls',
          'fav-pessoptomist',
          'fav-kakojkjk',
          'fav-medicinebleu',
          'fav-mellydoll',
          'fav-acleneave',
          'fav-shelleydeluca',
          'fav-dixproject',
          'fav-david_silva',
          'fav-possibleproductions',
          'fav-shelbaby',
          'fav-headcharge',
          'fav-leriu',
          'fav-oorlab',
          'fav-bebbesarc',
          'fav-lady_in_blue',
          'fav-bellhead',
          'fav-rubyvermillion',
          'fav-eatraw',
          'fav-airstreamdriver',
          'fav-arcticbutterfly',
          'fav-misoom',
          'fav-a1moviestar'
        ],
        num_reviews: [7],
        subjectSorter: ['need keyword'],
        publicdate: ['2007-03-01T15:13:59Z']
      },
      _index: 'prod-i-006'
    },
    {
      _score: 20.881666,
      _type: '_doc',
      _id: 'reasons_why_1',
      _source: {
        description: [
          "Television manufacturing at RCA's New Jersey plant. Includes scenes of design, engineering and quality control. Great scenes of TVs with flickering test patterns."
        ],
        title: ['Reasons Why, The (Part I)'],
        downloads: [11844],
        mediatype: ['movies'],
        collection: [
          'prelinger',
          'fav-creaturefan',
          'fav-mustafa_a_bhagat',
          'fav-kliffee',
          'fav-losflanagan',
          'fav-erika_colby',
          'fav-deep_pockets',
          'fav-eighty_bug',
          'fav-mistercastro',
          'fav-slbenz',
          'fav-lebeau11',
          'fav-vicki68',
          'fav-graidenb',
          'fav-thegreenflyingkiwi',
          'fav-ksharter',
          'fav-cd_smith',
          'fav-ueb',
          'fav-martin_felsenfeld',
          'fav-circularvortex',
          'fav-nostalgistudio',
          'fav-hesher',
          'fav-phiftybuick',
          'fav-david_silva',
          'fav-super8',
          'fav-tatiara7',
          'fav-lobotomy_club',
          'fav-mellydoll',
          'fav-edwardkildow',
          'fav-jimimod',
          'fav-bob_rhubart',
          'fav-mheinrichskd0udi',
          'fav-kevinsummers',
          'fav-sethry',
          'fav-korcuska'
        ],
        num_reviews: [7],
        subjectSorter: ['need keyword'],
        publicdate: ['2003-05-27T12:02:35Z']
      },
      _index: 'prod-i-006'
    },
    {
      _score: 20.84214,
      _type: '_doc',
      _id: 'along_the_way',
      _source: {
        creatorSorter: ['levene (carol)'],
        description: [
          'Depicts actual construction progress throughout the San Francisco Bay Area Rapid Transit System during 1967-68, emphasizing the original and continuing aims of the district--to provide safe, fast, economical and comfortable mass transit, thereby eliminating traffic congestion and preserving the beauty and character of the area.'
        ],
        title: ['Along the Way'],
        downloads: [7755],
        mediatype: ['movies'],
        collection: [
          'prelinger',
          'fav-david_silva',
          'fav-davykurlander',
          'fav-thomas_lopez794',
          'fav-breudd',
          'fav-meateye',
          'fav-smardzz',
          'fav-fruitiebat',
          'fav-project_greenpark',
          'fav-foolishblue',
          'fav-jordi_mumma',
          'fav-xeconya',
          'fav-thevelvetundergrad',
          'fav-thegreenflyingkiwi',
          'fav-chriscousin',
          'fav-margaret_b_1',
          'fav-iowatvman',
          'fav-blake_rand'
        ],
        num_reviews: [4],
        subjectSorter: ['need keyword'],
        publicdate: ['2003-05-16T10:39:21Z']
      },
      _index: 'prod-i-006'
    }
  ];
  const tree = renderer
    .create(
      <Router>
        <RelatedItemsSection relatedItems={relatedItems} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
