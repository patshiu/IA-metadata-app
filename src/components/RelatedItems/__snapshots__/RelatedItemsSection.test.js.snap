// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders related items section with several items correctly 1`] = `
<div
  className="row"
>
  <div
    className="section"
  >
    <h2>
      Related Items
       
      (3)
    </h2>
    <div
      className="related-cards-container"
    >
      <div
        className="related-item-card-container col-1-3 padded"
      >
        <div
          className="related-item-card"
        >
          <img
            alt="Thumbnail for To The Fair (Pt 2)"
            className="card-hero"
            src="https://archive.org/services/img/ToTheFairB"
          />
          <div
            className="card-content"
          >
            <h3>
              <strong>
                <a
                  href="/ToTheFairB"
                  onClick={[Function]}
                >
                  To The Fair (Pt 2)
                </a>
              </strong>
            </h3>
            <p>
              Media type: 
              movies
            </p>
            <p>
              Provides a trip to the New York World's Fair of 1965, showing the various activities at the fair.
            </p>
          </div>
        </div>
      </div>
      <div
        className="related-item-card-container col-1-3 padded"
      >
        <div
          className="related-item-card"
        >
          <img
            alt="Thumbnail for Reasons Why, The (Part I)"
            className="card-hero"
            src="https://archive.org/services/img/reasons_why_1"
          />
          <div
            className="card-content"
          >
            <h3>
              <strong>
                <a
                  href="/reasons_why_1"
                  onClick={[Function]}
                >
                  Reasons Why, The (Part I)
                </a>
              </strong>
            </h3>
            <p>
              Media type: 
              movies
            </p>
            <p>
              Television manufacturing at RCA's New Jersey plant. Includes scenes of design, engineering and quality control. Great scenes of TVs with flickering test patterns.
            </p>
          </div>
        </div>
      </div>
      <div
        className="related-item-card-container col-1-3 padded"
      >
        <div
          className="related-item-card"
        >
          <img
            alt="Thumbnail for Along the Way"
            className="card-hero"
            src="https://archive.org/services/img/along_the_way"
          />
          <div
            className="card-content"
          >
            <h3>
              <strong>
                <a
                  href="/along_the_way"
                  onClick={[Function]}
                >
                  Along the Way
                </a>
              </strong>
            </h3>
            <p>
              Media type: 
              movies
            </p>
            <p>
              Depicts actual construction progress throughout the San Francisco Bay Area Rapid Transit System during 1967-68, emphasizing the original and continuing aims of the district--to provide safe, fast, economical and comfortable mass transit, thereby eliminating traffic congestion and preserving the beauty and character of the area.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders review section with no reviews correctly 1`] = `
<div
  className="row"
>
  <div
    className="section"
  >
    <h2>
      Related Items
       
    </h2>
    <p>
      No related items.
    </p>
  </div>
</div>
`;

exports[`renders review section with undefined reviews object correctly 1`] = `
<div
  className="row"
>
  <div
    className="section"
  >
    <h2>
      Related Items
       
    </h2>
    <p>
      No related items.
    </p>
  </div>
</div>
`;
